cpp/tensorrt_llm/batch_manager/cacheTransceiver.cpp

```diff
@@ -343,9 +343,19 @@ void CacheTransceiver::checkContextTransferStatus(bool blocking)
             toCompleteIdSet.insert(requestId);
         }
     }
+
+    size_t idx = 0;
+    for (auto it = mResponderFutures.begin();
+         atLeastRequestNum.value_or(0) > static_cast<int>(toCompleteIdSet.size()) && it != mResponderFutures.end();
+         ++it, ++idx)
+    {
+        toCompleteIdSet.insert(it->first->mRequestId);
+    }
```

Comment:
This was not part of the original closed source MR. Why is this needed?

Reply:
I restructured the MR according to @chuangz0 comment to wait on the futures according to the insertion order.